src/guiVolumeChange.cpp

```diff
@@ -136,29 +145,50 @@ bool GUIVolumeChange::OnEvent(const SEvent& event)
 			quitMenu();
 			return true;
 		}
-	}
-
-	if (event.GUIEvent.EventType == gui::EGET_BUTTON_CLICKED) {
-		if (event.GUIEvent.Caller->getID() == ID_soundExitButton) {
-			quitMenu();
+	} else if (event.EventType == EET_GUI_EVENT) {
+		if (event.GUIEvent.EventType == gui::EGET_CHECKBOX_CHANGED) {
+			gui::IGUIElement *e = getElementFromId(ID_soundMuteButton);
+			if (e != NULL && e->getType() == gui::EGUIET_CHECK_BOX) {
+				g_settings->setBool("mute_sound", ((gui::IGUICheckBox*)e)->isChecked());
+			}
+
+			Environment->setFocus(this);
 			return true;
 		}
-	}
 
-	if (event.GUIEvent.EventType == gui::EGET_SCROLL_BAR_CHANGED) {
-		if (event.GUIEvent.Caller->getID() == ID_soundSlider) {
-			s32 pos = ((gui::IGUIScrollBar*)event.GUIEvent.Caller)->getPos();
-			g_settings->setFloat("sound_volume", (float) pos / 100);
-
-			gui::IGUIElement *e = getElementFromId(ID_soundText);
-			const wchar_t *text = wgettext("Sound Volume: ");
-			core::stringw volume_text = text;
-			delete [] text;
+		if (event.GUIEvent.EventType == gui::EGET_BUTTON_CLICKED) {
+			if (event.GUIEvent.Caller->getID() == ID_soundExitButton) {
+				quitMenu();
+				return true;
+			}
+			Environment->setFocus(this);
+		}
 
-			volume_text += core::stringw(pos) + core::stringw("%");
-			e->setText(volume_text.c_str());
-			return true;
+		if (event.GUIEvent.EventType == gui::EGET_ELEMENT_FOCUS_LOST
+				&& isVisible()) {
```

Comment:
`GUIMainMenu: Not allowing focus change.` is thrown for each change. So, should this probably be `!isVisible()`?
Also, why is this check part at all?

Reply:
https://github.com/minetest/minetest/blob/d8f20d029a17551dab35a850bed48a62c1a02618/src/guiKeyChangeMenu.cpp#L340-L350
It's the same as here, but I agree that the message could be removed.
Without it, the focus gets lost if you click outside the form and the escape key event no longer activates.